core/interfaces/branch.py

```diff
@@ -524,43 +534,36 @@ def get_selected_branch_name(self, selection, line):
         branch_name = segments[1]
         local_region = self.view.get_regions("git_savvy_interface.branch_list")[0]
         if local_region.contains(selection):
-            return branch_name
+            return (None, branch_name)
 
         remotes = self.get_remotes()
         for remote_name in remotes:
             remote_region = self.view.get_regions("git_savvy_interface.branch_list_" + remote_name)
             if remote_region and remote_region[0].contains(selection):
-                return "{}/{}".format(remote_name, branch_name)
-
+                return (remote_name, branch_name)
 
-class GsBranchesFetchAndMergeCommand(TextCommand, GitCommand):
 
+class GsBranchesFetchAndMergeCommand(GsBranchesMergeSelectedCommand):
     """
     Fetch from remote and merge fetched branch into active branch.
     """
 
-    def run(self, edit):
-        sublime.set_timeout_async(self.run_async, 0)
-
     def run_async(self):
         self.interface = ui.get_interface(self.view.id())
-        selection, line = self.interface.get_selection_line()
-        if not line:
-            return
 
-        segments = line.strip("▸ ").split(" ")
-        branch_name = segments[1]
+        branches = self.get_selected_branches()
+        branches_strings = set()
+        for branch in branches:
```

Comment:
This for loop sort of repeats 508 doesn't it? Can we factor these out into a helper method perhaps? I.e a sort of `get_branch_names_with_remote` that can be used in both places? Calling `self.fetch` can be toggled with a boolean argument